src/CustomComposerHelper.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace JustCoded\WP\Composer;
+
+use Composer\IO\IOInterface;
+use Composer\Script\Event;
+
+/**
+ * Works with composer data
+ *
+ * Class CustomComposerHelper
+ *
+ * @package JustCoded\WP\Composer
+ */
+class CustomComposerHelper {
+
+	/**
+	 * Help to pass arguments to array
+	 *
+	 * @param array $arguments
+	 *
+	 * @return array
+	 */
+	public static function arguments_cleaner( $arguments = array() ) {
+
+		$args_ready = array();
+		foreach ( $arguments as $key => $arg_str ) {
+			$one_arg_pair = explode( '=', $arg_str );
+			if ( count( $one_arg_pair ) > 1 ) {
+				$key_of_param = 'default';
+				switch ( $one_arg_pair[0] ) {
+					case '-t' :
+						$key_of_param = 'title';
+						break;
+					case '-ns' :
+						$key_of_param = 'namespace';
+						break;
+					case '-dir' :
+						$key_of_param = 'dir';
+						break;
+				}
+				$args_ready[ $key_of_param ] = $one_arg_pair[1];
+			} else {
+				if ( '-s' === $one_arg_pair[0] ) {
+					$args_ready['silent'] = true;
+				} else {
+					$args_ready['silent'] = false;
+					$args_ready['theme_slug'] = $one_arg_pair[0];
+				}
+			}
+		}
```

Comment:
the parser is hard-coded. You can't use it for other commands in this way.

Parser should be smart and analyze the command by itself and break into key/values.

Like any console script it should understand arguments like: '-t something', '-t=something'. And you should not hard-code these keys.